### PR TITLE
Implement forms generation

### PR DESCRIPTION
**Pull request by alekseyzadorozhniy**
Implementation of the forms generation from domain schemas

### CODE REVIEW

File: packages/formik/src/components/FieldAdapter.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+export default class Field extends Component<Props, {}> {
+  public static contextTypes = {
+    formik: PropTypes.object
+  };
+
+  constructor(public props: Props, public context: any) {
+    super(props, context);
+    if (!context.formik) {
```
Comment (evgenijolshanskij) [inline packages/formik/src/components/FieldAdapter.tsx]: Do we really need this check here? We don't allow users to create a form manually.

Comment (alekseyzadorozhniy) [reply packages/formik/src/components/FieldAdapter.tsx]: It's not necessary. I plan to clean up all components later.

File: packages/validation/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export interface SchemaContext {
+  fieldName: string;
+  schema: any;
+  values: any;
+}
+
+export type Value = string | number | boolean;
+
+export type Validator = string | undefined;
```
Comment (evgenijolshanskij) [inline packages/validation/src/types.ts]: This export use none of the other files. Do we really need it?

Comment (alekseyzadorozhniy) [reply packages/validation/src/types.ts]: You are right.

File: packages/formik/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            model[fieldName] = {};
+            getValues(schema[fieldName].type.values, model[fieldName]);
+          }
+        });
+      return model;
+    };
+    return getValues(this.schema.values, {});
+  }
+
+  private genInput(ctx: any, value: string | number, fieldName: string, parent: any) {
```
Comment (evgenijolshanskij) [inline packages/formik/src/index.tsx]: All of those Field components have a common structure. Why do we need 5 similar methods to generate the same component?

File: packages/formik/src/__tests__/index.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+import DomainReactForms, { FieldTypes } from '../';
+
+describe('DomainFormik', () => {
+  it('should generate simple form', () => {
+    const schema = new DomainSchema(
+      class extends Schema {
+        public __ = { name: 'User' };
+        public id = DomainSchema.Int;
+        public buttons = {
+          type: DomainReactForms.FormButtons,
```
Comment (evgenijolshanskij) [inline packages/formik/src/__tests__/index.spec.tsx]: Why a component type in plural?

File: packages/formik/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,202 @@
+import DomainSchema from '@domain-schema/core';
+import DomainValidator from '@domain-schema/validation';
+import { FormikProps, withFormik } from 'formik';
+import * as React from 'react';
+import { Button, Field, Form, RenderCheckBox, RenderField, RenderRadio, RenderSelect } from './components';
+import FieldTypes from './fieldTypes';
+
+export default class DomainReactForms {
+  public static FormButtons = class FormButtons {};
```
Comment (evgenijolshanskij) [inline packages/formik/src/index.tsx]: What is the point to use the class?

File: packages/formik/README.md [hunk 6]
```diff
@@ -30,46 +61,392 @@ const userFormSchema = new DomainSchema(
...
+    submit: {
+      label: 'Submit',
+      className: 'submit-btn',
+      color: 'primary'
+    },
+    reset: {
+      label: 'Reset',
+      className: 'reset-btn'
+    },
+    reverse: true
```
Comment (evgenijolshanskij) [inline packages/formik/README.md]: It is not obvious.

File: packages/formik/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+        .forEach((fieldName: string) => {
+          if (fieldName === 'id') {
+            return;
+          }
+          const schemaField = schema[fieldName];
+          if (!schemaField.type.isSchema) {
+            model[fieldName] = schemaField.defaultValue || '';
+          } else {
+            model[fieldName] = {};
+            getValues(schema[fieldName].type.values, model[fieldName]);
```
Comment (evgenijolshanskij) [inline packages/formik/src/index.tsx]: The `getValues` function returns a result, so you cannot invoke it just like this. You must assign `getValues` result value to the model[fieldName].

Comment (alekseyzadorozhniy) [reply packages/formik/src/index.tsx]: Ok

File: packages/formik/src/__tests__/index.spec.tsx (hunk 4 above)
Comment (alekseyzadorozhniy) [reply packages/formik/src/__tests__/index.spec.tsx]: I plan to redesign form buttons

File: packages/formik/src/index.tsx (hunk 5 above)
Comment (alekseyzadorozhniy) [reply packages/formik/src/index.tsx]: I plan to redesign form buttons

File: packages/formik/src/index.tsx (hunk 3 above)
Comment (alekseyzadorozhniy) [reply packages/formik/src/index.tsx]: Ok, I'll figure out how to improve it

**Pull request merged**